### PR TITLE
Add release jobs for namespace-controller and its tests

### PR DESCRIPTION
**Pull request by sjanota**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

Implements #193 
Implements #348

### CODE REVIEW

File: prow/jobs/kyma/tests/test-namespace-controller/test-namespace-controller.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
+test_infra_ref: &test_infra_ref
+  org: kyma-project
+  repo: test-infra
+  path_alias: github.com/kyma-project/test-infra
+
 job_template: &job_template
+  name: kyma-tests-test-namespace-controller
   skip_report: true
-  run_if_changed: "^tests/test-namespace-controller/"
```
Comment (aszecowka) [inline prow/jobs/kyma/tests/test-namespace-controller/test-namespace-controller.yaml]: skip_report: false <- right now we disable Jenkins and make Prow official CI, it is not yet documented ;)

File: prow/jobs/kyma/components/namespace-controller/namespace-controller.yaml [hunk 2]
```diff
@@ -1,23 +1,23 @@
+test_infra_ref: &test_infra_ref
+  org: kyma-project
+  repo: test-infra
+  path_alias: github.com/kyma-project/test-infra
+
 job_template: &job_template
+  name: kyma-components-namespace-controller
   skip_report: true
```
Comment (aszecowka) [inline prow/jobs/kyma/components/namespace-controller/namespace-controller.yaml]: skip_report: false <-- see my other comment

File: development/tools/jobs/kyma/namespace_controller_test.go [hunk 3]
```diff
@@ -41,23 +35,42 @@ func TestNamespaceControllerJobPostsubmit(t *testing.T) {
...
+			require.NoError(t, err)
+			actualPresubmit := tester.FindPresubmitJobByName(jobConfig.Presubmits["kyma-project/kyma"], "kyma-components-namespace-controller", currentRelease)
+			require.NotNil(t, actualPresubmit)
+			assert.True(t, actualPresubmit.SkipReport)
+			assert.True(t, actualPresubmit.Decorate)
+			assert.Equal(t, "github.com/kyma-project/kyma", actualPresubmit.PathAlias)
+			tester.AssertThatHasExtraRefTestInfra(t, actualPresubmit.JobBase.UtilityConfig, currentRelease)
+			tester.AssertThatHasPresets(t, actualPresubmit.JobBase, tester.PresetDindEnabled, tester.PresetDockerPushRepo, tester.PresetGcrPush, tester.PresetBuildRelease)
+			assert.True(t, actualPresubmit.AlwaysRun)
+			assert.Len(t, actualPresubmit.JobBase.Spec.Containers, 1)
```
Comment (kubadz) [inline development/tools/jobs/kyma/namespace_controller_test.go]: You can use tester.AssertThatExecGolangBuidlpack here, but it is not mandatory.

File: development/tools/jobs/kyma/test_namespace_controller_test.go [hunk 4]
```diff
@@ -42,23 +35,43 @@ func TestNamespaceControllerTestsJobPostsubmit(t *testing.T) {
...
+			require.NoError(t, err)
+			actualPresubmit := tester.FindPresubmitJobByName(jobConfig.Presubmits["kyma-project/kyma"], "kyma-tests-test-namespace-controller", currentRelease)
+			require.NotNil(t, actualPresubmit)
+			assert.True(t, actualPresubmit.SkipReport)
+			assert.True(t, actualPresubmit.Decorate)
+			assert.Equal(t, "github.com/kyma-project/kyma", actualPresubmit.PathAlias)
+			tester.AssertThatHasExtraRefTestInfra(t, actualPresubmit.JobBase.UtilityConfig, currentRelease)
+			tester.AssertThatHasPresets(t, actualPresubmit.JobBase, tester.PresetDindEnabled, tester.PresetDockerPushRepo, tester.PresetGcrPush, tester.PresetBuildRelease)
+			assert.True(t, actualPresubmit.AlwaysRun)
+			assert.Len(t, actualPresubmit.JobBase.Spec.Containers, 1)
```
Comment (kubadz) [inline development/tools/jobs/kyma/test_namespace_controller_test.go]: You can use tester.AssertThatExecGolangBuidlpack here, but it is not mandatory.

**Pull request merged**